lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/api/VortexAggregateFuture.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.vortex.api;
+
+import org.apache.commons.lang3.NotImplementedException;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.reef.annotations.Unstable;
+import org.apache.reef.annotations.audience.ClientSide;
+import org.apache.reef.annotations.audience.Private;
+import org.apache.reef.annotations.audience.Public;
+import org.apache.reef.vortex.common.VortexFutureDelegate;
+import org.apache.reef.vortex.driver.VortexMaster;
+import org.apache.reef.wake.remote.Codec;
+
+import javax.annotation.concurrent.NotThreadSafe;
+import java.util.*;
+import java.util.concurrent.*;
+import java.util.logging.Logger;
+
+/**
+ * The interface between user code and aggregation Tasklets.
+ * Thread safety: This class is not meant to be used in a multi-threaded fashion.
+ * TODO[JIRA REEF-1131]: Create and run tests once functional.
+ */
+@Public
+@ClientSide
+@NotThreadSafe
+@Unstable
+public final class VortexAggregateFuture<TInput, TOutput> implements VortexFutureDelegate {
```

Comment:
My primary concern is the order at which results are put into the queue -- if some of the methods are not synchronized, a function that returns "later" preempts an "earlier" function, the `hasNext` field in `AggregateResult` would be incorrect. It also reduces the errors that the user can make when using the API, since this was not meant to be accessed in a multi-threaded fashion anyhow. Provided that we're not doing any intensive work here, I thought it would be okay to mark the functions as `synchronized` to better ensure correctness.